src/main/java/org/springframework/data/elasticsearch/core/ReactiveDocumentOperations.java

```diff
@@ -78,6 +86,74 @@
 	 */
 	<T> Mono<T> save(T entity, IndexCoordinates index);
 
+	/**
+	 * Index entities under the given {@literal type} in the given {@literal index}. If the {@literal index} is
+	 * {@literal null} or empty the index name provided via entity metadata is used. Same for the {@literal type}.
+	 *
+	 * @param entities must not be {@literal null}.
+	 * @param index the target index, must not be {@literal null}
+	 * @param <T>
+	 * @return a {@link Flux} emitting saved entities.
+	 */
+	default <T> Flux<T> saveAll(Iterable<T> entities, IndexCoordinates index) {
+		return saveAll(Flux.fromIterable(entities), index);
+	}
+
+	/**
+	 * Index entities under the given {@literal type} in the given {@literal index}. If the {@literal index} is
+	 * {@literal null} or empty the index name provided via entity metadata is used. Same for the {@literal type}.
+	 *
+	 * @param entities must not be {@literal null}.
+	 * @param index the target index, must not be {@literal null}
+	 * @param <T>
+	 * @return a {@link Flux} emitting saved entities.
+	 */
+	<T> Flux<T> saveAll(Publisher<T> entities, IndexCoordinates index);
```

Comment:
`saveAll` should accept `Mono<? extends Collection<? extends T>> batchToSave` instead of `Publisher<T>`. Behind the scenes, we're buffering objects until the stream completes and one does not see from the method declaration when the save happens (on each element, on completion, on some buffer size). Using `Mono` with a collection offloads the batch size onto the caller and we express the single-request nature of the method.